pkg/ddl/session/session_pool.go

```diff
@@ -34,11 +35,11 @@ type Pool struct {
 		sync.Mutex
 		closed bool
 	}
-	resPool *pools.ResourcePool
+	resPool util.SessionPool
 }
 
 // NewSessionPool creates a new Session pool.
-func NewSessionPool(resPool *pools.ResourcePool) *Pool {
+func NewSessionPool(resPool util.SessionPool) *Pool {
```

Comment:
in nextgen, as we are not registering this cross ks session into the SessionManager, so it will not block GC during scanning kv?

Reply:
Yes, it is a risk. Not sure if the GC worker can execute across keyspaces.